temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java

```diff
@@ -287,4 +372,72 @@ public void setBeanFactory(@Nonnull BeanFactory beanFactory) throws BeansExcepti
     Assert.isInstanceOf(ConfigurableListableBeanFactory.class, beanFactory);
     this.beanFactory = (ConfigurableListableBeanFactory) beanFactory;
   }
+
+  private static class Workers {
+    private final Map<String, Worker> workersByName = new HashMap<>();
+    private final Map<String, Worker> workersByTaskQueue = new HashMap<>();
+    private final List<Worker> workers = new ArrayList<>();
+
+    public void addWorker(WorkerDescriptor workerDescriptor) {
+      Worker newWorker = workerDescriptor.getWorker();
+      Worker existingWorker = workersByName.get(workerDescriptor.getName());
+      if (existingWorker != null
+          && !existingWorker.getTaskQueue().equals(newWorker.getTaskQueue())) {
```

Comment:
Shouldn't name be globally unique regardless of task queue?

Reply:
It's to catch a situation when someone gives a worker an explicit name of ANOTHER worker's task queue.